### DISCUSSION TITLE
fire extinguishing grenade

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#12779] [Ideas and suggestions]
[this thing actually exists in reality](https://www.youtube.com/watch?v=oOwOYVJ9IgA)

massive fire can be difficult to extinguish so it would be nice to have fire extinguishing grenade along with fix foam grenade

i guess it's easy to implement: just change fix foam grenade particles to extinguisher particles
but for now i'm too lazy to do that

**Comment by Next-Door-Tech** [comment]
Options to consider:

1. Add fire extinguishing as an additional effect of the current fixfoam grenade.
   - Upside: perhaps makes the most sense real-world wise. Fixfoam grenades are currently crafted with 1 steel + 1 sodium + 1 phosphorus, real [dry powder fire extinguishers](https://en.wikipedia.org/wiki/Fire_extinguisher#Dry_chemical) use either sodium bicarbonate [NaHCO<sub>3</sub>] and/or monoammonium phosphate [NH<sub>4</sub>H<sub>2</sub>PO<sub>4</sub>]. I assume the grenades in the OP's link use one of these two chemicals.
   - Downside: I worry this may make fixfoam too OP, so I don't necessarily like this for interesting gameplay.
2. Add separate extinguishing grenades made with 1 iron/steel + 1 sodium + 1 phosphorus.
   - Upside: Most similar to current game balance, and can still make real-world sense. Extinguishing grenade recipe chosen using reasoning above.
   - Downside: Recipe would be (near) identical to the current fixfoam grenade recipe, begging the question why fixfoam grenades don't also extinguish fires.
3. Separate extinguishing grenades like (2), rework fixfoam recipe.
   - Alternate fixfoam recipe ideas:
      - 1 steel + 1 elastin (as an adhesive) + 1 phosphorus (idk feels chemical-y in the right way)
      - 1 steel + 1 plastic (as a crack-filling material) + 1 alcohol (stand-in for a different solvent like acetone that actually melts plastic)
   - Upside: Balanced like 2, removes recipe clash and new fixfoam recipe makes more real-world sense imo
   - Downside: Have to change existing fixfoam grenade recipe
4. Same as 3, but make fixfoam grenade gas _highly flammable_
   - Upside: Same as 3, but definitely more exciting. I feel this would create some fun scenarios to deal with in-game.
   - Downside: Have to add new flammability mechanics for fixfoam as well as change recipe
5. Leave fixfoam grenade as-is (or change like 3 or 4, doesn't matter), and add [halon gas-based](https://en.wikipedia.org/wiki/Fire_extinguisher#Halons,_Halon-replacement_clean_agents_and_carbon_dioxide) extinguishing grenades.
   - Halon grenade recipe might be 1 iron/steel + 1 carbon + 2 chlorine. Actual halon gas extinguishing agents use chemicals made of carbon and some combination of fluorine, chlorine, and bromine, though two of those elements don't exist in-game and I see no reason to add them only for one recipe.
   - Halon gas is extremely toxic, so this effect should be modeled in game.
   - Upside: Balanced, plus halon gas seems to fit the game's thematic use of somewhat dangerous though very effective substances in a desperate environment (opiates, methamphetamine, haloperidol, etc.)
   - Downside: Have to implement halon gas as a toxic, fire-extinguishing gas, which likely will require new code. (Though, the wiki mentions toxic gas as an effect the coelacanth was meant to have, so perhaps this code already exists in part?)

My personal favorites are options 4 and 5, or both the flammable fixfoam from 4 and the halide extinguishers from 5 combined, though I realize these will require the most work.

Thoughts?

**Comment by SomeRandomNoobKekeke** [comment]
Now that i thought about it, i don't see much use
1. Fire on submarine 
you can just [override pumps and extinguish it with water ](https://www.youtube.com/watch?v=BX0QA995aBw)
2. Fire on outposts
if you ignored [this event](https://barotraumagame.com/wiki/Events#Firefighting), it's totally your fault and you should burn to death for that
3. Fighting ballast flora
you can just close hatch and wait for fire to burn all oxygen, if there's no hatch you can make some scheme and turn pump on from a distance

the only problem is fire on Kastrull upper mid, because you need to run around fire to get extinguisher, or flood the sub completely 
![kastrull fire](https://github.com/Regalis11/Barotrauma/assets/122838333/8b854527-6310-439d-b3b7-afe46baa0519)

but it's not really a problem because you can store extinguishers like that
![20231012115117_1](https://github.com/Regalis11/Barotrauma/assets/122838333/81bdb3a7-4364-49ac-8c19-0a386d1c9d42)
it would be a problem if you couldn't

option 1 would be logical if incendium grenade would also do structural damage, then you could use it to clean up all consequences
 
option 4, if you're throwing fix grenade then there's probably hull breach and a lot of water
option 5, you won't gonna stand in fire and inhale gas anyway, it'll be more like chemical weapon

i think it would work only for [noobs that can't comprehend wiring](https://www.youtube.com/watch?app=desktop&v=sIxKI7o_S0E)

**Reply by Next-Door-Tech** [reply]
Some fair points.

> Now that i thought about it, i don't see much use

I agree that this isn't a _strictly necessary_ addition, we already have handheld fire extinguishers and we can always flood the sub as you mention. That said, I still think extinguishing grenades would be a cool additional option to have.

As a counterexample where an extinguishing grenade would have been more useful than an alternative, here is a scenario that happened to me on my singleplayer save. My captain was trapped on the bridge by a ballast flora fire that got out of control, and the fire had spread through the medbay. The only storage I could reach without running through flames was the supply cabinet on the bridge.

![Orca2 Captain Trapped on Bridge](https://github.com/Regalis11/Barotrauma/assets/5867663/52d1e8d8-1c2a-4e58-a038-ed8ed2503daa)

At the time I had to control a different crew member and put out the fire with an extinguisher from the other end, but that meant I spent quite a while without a human at the helm. I now keep an extinguisher in that cabinet, but a grenade in that same scenario would mean the captain could return to the helm even more quickly.

> option 1 would be logical if incendium grenade would also do structural damage, then you could use it to clean up all consequences

Option 1 was really only there because it would take the least work to implement. I still think it's too powerful.

> option 4, if you're throwing fix grenade then there's probably hull breach and a lot of water

True, hadn't thought about that tbh. Although, there might still be a scenario where there's a small lingering fire and a somewhat minor hull breach that hasn't completely flooded the compartment. I still like the idea of a player not reading the warning in the fixfoam grenade description and learning an explosive lesson, but you are right that this would be an edge case.

> option 5, you won't gonna stand in fire and inhale gas anyway, it'll be more like chemical weapon

Doing some more reading on Halon, it looks like its actual toxicity is really only a problem in long-term, low dose exposure. However, because it does its job by displacing oxygen, death by suffocation is the primary short-term concern. IMO the Halon gas should linger in the room for a while (idk, 60s?) and induce oxygen low in crew not wearing a breathing apparatus.

**Comment by SomeRandomNoobKekeke** [comment]
Actually not so easy to implement
Extinguishing seems to be linked to extinguisher as RepairTool and not to the particles
In theory we can put little extinguisher inside grenade and turn it on when grenade is thrown :D

![barodev](https://github.com/Regalis11/Barotrauma/assets/122838333/600d797b-849d-4205-8764-7156157c39ec)

**Comment by SomeRandomNoobKekeke** [comment]
I managed to make throwable fire extinguisher that activates when onFire
It works very clumsily because extinguishing isn't area effect, it's ray effect, so it works only in one direction
also if fire is extinguished by flying extinguisher  game crashes

[link to mod if you want to try it](https://steamcommunity.com/sharedfiles/filedetails/?id=3050691412)

**Comment by gerakis** [comment]
It would make detonators more useful. Right now they are really only useful for griefing or traps for sub/station builders. Picking up a detonator is one of those "griefer detector" things I look for in the server logs.

This could probably be added to fixfoam grenades. Just make them also extinguish fires. They currently arent useful enough in my opinion.
